libevmasm/RuleList.h

```diff
@@ -694,4 +698,32 @@ std::vector<SimplificationRule<Pattern>> simplificationRuleList(
 	return rules;
 }
 
+/// @returns a list of generic simplification rules with additional rules that may apply for the given dialect.
+template <class Pattern>
+std::vector<SimplificationRule<Pattern>> simplificationRuleList(
+	yul::Dialect const& _dialect,
+	Pattern A,
+	Pattern B,
+	Pattern C,
+	Pattern W,
+	Pattern X,
+	Pattern Y,
+	Pattern Z
+)
+{
+	std::vector<SimplificationRule<Pattern>> rules{simplificationRuleList(A, B, C, W, X, Y, Z)};
```

Comment:
Why don't you change that function instead?

Reply:
I thought it may make sense to separate evm version specific rules from generic rules

Reply:
In a way, we already have a lot of them mixed in, but they do not matter because they do not introduce a new opcode.

Reply:
Ah oki, I will change the function.